### PR TITLE
Add Golangci-lint to GitHub Actions and add Deadcode linter

### PR DESCRIPTION
**Pull request by estensen**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

Improve code quality

**Which issues(s) does this PR fix?**

https://github.com/prysmaticlabs/prysm/issues/9371

**Other notes for review**

One linter at a time will be added at a time instead of adding all the linters all at once to make review easier.

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go [hunk 1]
```diff
@@ -496,6 +496,7 @@ func (vs *Server) inRangeVotes(ctx context.Context,
 	return inRangeVotes, nil
 }
 
+//nolint:deadcode
```
Comment (rkapka) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go:499]: We can remove this function.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -35,6 +35,12 @@ jobs:
         uses: securego/gosec@master
         with:
           args: '-exclude-dir=shared/bls/herumi ./...'
+
+      - name: Golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          args: --print-issued-lines --sort-results --disable-all -E deadcode --skip-dirs=validator/testing --skip-files=validator/web/site_data.go
```
Comment (rkapka) [inline .github/workflows/go.yml:42]: `validator/testing` has regular code in it, and as such should also be linted.

**Review by rkapka — CHANGES_REQUESTED** [review]

**Review by rkapka — DISMISSED** [review]
Thanks for doing this!

**Review by prestonvanloon — DISMISSED** [review]

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
@estensen The `Go / Check` action fails because of a timeout:

> level=error msg="Timeout exceeded: try increasing it by passing --timeout option"

We will have to increase it. I wonder how long it takes. Hopefully not too long. Do you have any data?

**Comment by estensen** [comment]
@rkapka running it locally on my 16" MacBook Pro after clearing the golangci-lint cache it takes 16 seconds, so I think it will only take slightly more than the default 1m timeout.
I'm increasing the timeout to 15m for now.

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
Hey @estensen , can you please fix conflicts? Thanks!

**Review by terencechain — APPROVED** [review]

**Comment by rkapka** [comment]
Hey @estensen , can you please redo this?

> Please run bazel run //:gazelle -- fix and commit the BUILD.bazel file(s), otherwise the linter CI check will fail.

I apologize that it takes so long merge this PR.

**Comment by estensen** [comment]
haha, no worries

**Comment by rkapka** [comment]
@estensen There are build issues in `beacon-chain/slasher/helpers.go`

**Review by rkapka — DISMISSED** [review]

**Review by rkapka — DISMISSED** [review]

**Comment by rkapka** [comment]
`config/params/loader_test.go` has some issues now @estensen

**Comment by rauljordan** [comment]
Hi @estensen apologies for the delay on this, but you caught us at a difficult time when the codebase is shifting a lot as we are preparing a v2.0.0 release. We are now done with picking our release candidate and your PR wont accumulate as many conflicts as before. Feel free to ping me whenever it's ready again and we'll approve it quickly

**Comment by estensen** [comment]
@rauljordan haha, this feels like a game of catch-up

PTAL

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
> @rauljordan haha, this feels like a game of catch-up

Congratulations! You've won :-)

**Comment by prestonvanloon** [comment]
Glad to see this finally merged. Thanks @estensen for your contribution and patience!

**Pull request merged**